3-UsingAPIs/Week1/homework/ex4-pokerDiceAll.js

```diff
@@ -19,8 +19,15 @@ const rollDice = require('../../helpers/pokerDiceRoller');
 
 function rollTheDices() {
   // TODO Refactor this function
-  const dices = [1, 2, 3, 4, 5];
-  return rollDice(1);
+  return new Promise((resolve)=>{
+    const dices = [1, 2, 3, 4, 5];
+    const newArray=dices.map(el => rollDice(el))
+    const allPromises = Promise.all(newArray)
+    resolve(allPromises)
+  })
```

Comment:
Good progress so far. There are some issue with this implementation.
The main issue is that you resolve a promise, that's usually not what we want to do. We usually resolve some specific value like `resolve(1)` , `resolve("hello")`, `resolve([1,2,3])`

since `Promise.all` returns us a promise itself, we can just return it.


In addition, the `rollTheDices` function don't have to create it's own new promise as long as it returns a promise, it will work.